### ISSUE TITLE
Request tab no reappear 

### ISSUE THREAD

**Issue by MultiMadjik** [BobHasNoSoul/jellyfin-mods#41]
Hello 🙂 

First, congratulations to this awesome job!

Last day, I put the request tab button and it's works 
Sadly, the tab as no longer work when I switch to a library or another's in the website 
I do refresh the page to see again the request tab 

Can you have any ideas?

Thanks for advance!

**Comment by sureshfizzy** [comment]
+1  for this, I'm no longer able to login to my jellyseer inside the requests tab after setting it up. @ToMattBan any idea on this ?

**Comment by ToMattBan** [comment]
Good catch @MultiMadjik. I'm going to take a look at that. Never really had noticed it before.
Just to be sure, the tab don't show again when you click on "Home", coming back from one of your libraries, right?

About being unable to login, @sureshfizzy I tried some things here and it seems related to "max number of devices" setting.
I put it no -1 again and had no trouble logging in. Maybe clean the devices that are logged can help you?

**Comment by sureshfizzy** [comment]
Still not able to do that.. i checked on web and i get this, Am i missing something ?

myjellyseerip:port/api/v1/auth/me:1 

Failed to load resource: the server responded with a status of 401 ()

FYI: Jellyseer is working when i connect to it directly

**Comment by ToMattBan** [comment]
401 is an error code about unauthorized access.
For what I understood about how it works, the `/api/v1/auth/me` uses the cookies got from another request.
One request above, there is the `/api/v1/auth/jellyfin` request. This one return some error or does it login successfully?

**Comment by sureshfizzy** [comment]
surprisingly I couldn't find /api/v1/auth/jellyfin so i believe there are no errors on it. and regarding the 401, i'm able to login to my jellyseer from web though, it's just I can't login from the requests tab inside my jellyfin.

**Comment by MakD** [comment]
Same here @ToMattBan. After entering a library, it doesn't reappear,  and after pressing home as well.

**Comment by ToMattBan** [comment]
@sureshfizzy can you open another issue about the login stuff? Let's keep this one about the tab not reappearing. 

As soon as I got some (prob later today or tomorrow) time I will take a look on the issues.

**Comment by MakD** [comment]
@ToMattBan thanks for agreeing to take a look. Also, wanted to get this to your attention, this doesn't play well with the featured content bar, and also, doesn't seem to render well on a mobile device as well. 
![screen-20240921-040038-ezgif com-optimize](https://github.com/user-attachments/assets/d3f9d9df-358c-478c-bdc0-36938ad52603)

**Comment by sureshfizzy** [comment]
> @sureshfizzy can you open another issue about the login stuff? Let's keep this one about the tab not reappearing.
> 
> As soon as I got some (prob later today or tomorrow) time I will take a look on the issues.

Sure, thanks :)

**Comment by MakD** [comment]
Also @ToMattBan I had to set the width and height to vw and vh values as somehow the calc is messing the entire thing up (on desktop and phone both)

![Screenshot_20240921-111607](https://github.com/user-attachments/assets/ccd7e0b4-bbf8-49c5-b8fe-66acbff8d74e)

**Comment by ToMattBan** [comment]
@MakD Which browser/client are you using?
I tested here on my browser phone, jellyfin app, desktop and it doesn't shows like yours. The mobile here is great.


About the tab vanishing, @MultiMadjik @sureshfizzy I found why it was happening. I edited the instructions, can you test it please?

You can see the edited instructions here: 
https://github.com/ToMattBan/jellyfin-mods/blob/patch-1/10.9.x.md#add-requests-tab

It fixed for me, hope it fixes for you too

**Comment by MakD** [comment]
> @MakD Which browser/client are you using?
> I tested here on my browser phone, jellyfin app, desktop and it doesn't shows like yours. The mobile here is great.
> 
> 
> About the tab vanishing, @MultiMadjik @sureshfizzy I found why it was happening. I edited the instructions, can you test it please?
> 
> You can see the edited instructions here: 
> https://github.com/ToMattBan/jellyfin-mods/blob/patch-1/10.9.x.md#add-requests-tab
> 
> It fixed for me, hope it fixes for you too

@ToMattBan official android client. I changed the width calc to harcoded 98vw and that works for now. The only issue will be getting the reversed proxied url to open, which I have searched heaven and hell but couldn't get my hands around. Tried different caddy variation ls as well ( I'm not that client with caddy setup, so🤷). Only thing I came across is it needs a modded setup of jellyfin-web I guess. Even though I posted a query in the jellyseerr discord let's see if they address that.

**Comment by ToMattBan** [comment]
Thanks @MakD

**Comment by MakD** [comment]
@ToMattBan it's still disappearing for me

**Comment by ToMattBan** [comment]
@MakD did you clean the cache? If yes, in the index.html, before `const title` put a `console.log("Reaching this point")` to be sure that the script is really running when coming back to Home screen

**Comment by ToMattBan** [comment]
Gonna look at that @MakD 
I haven't test hitting the back button.
Seems Jellyfin have a lot of different behaviours, hard to catch them all. If you find another way to return to home that doesn't render the tab, please let me know

**Comment by MakD** [comment]
@ToMattBan please test this I guess solved it

`    <script>
      const createRequestTab = () => {
        console.log("Creating request tab");
        const title = document.createElement("div");
        title.classList.add("emby-button-foreground");
        title.innerText = "Requests";
  
        const button = document.createElement("button");
        button.type = "button";
        button.is = "empty-button";
        button.classList.add("emby-tab-button", "emby-button", "lastFocused");
        button.setAttribute("data-index", "2");
        button.setAttribute("id", "requestTab");
        button.appendChild(title);
  
        (function e() {
          const tabb = document.querySelector(".emby-tabs-slider");
          if (tabb && !document.querySelector("#requestTab")) {
            tabb.appendChild(button);
          } else if (!tabb) {
            setTimeout(e, 500);
          }
        })();
      };
  
      // Initial page load
      document.addEventListener("DOMContentLoaded", () => {
        createRequestTab();
      });
  
      // When navigating back or forward
      window.addEventListener("popstate", () => {
        createRequestTab();
      });
    </script>`

**Comment by ToMattBan** [comment]
hahaha yeah @MakD 
I lost some time trying tons of stuff and it just doesn't work.

Every way to come back to home have it's own behaviour, and this is making me crazy hahahaha.
Sometimes it re-render the page, sometimes it just changes the display property of a part of the page. 
Sometimes it just nuke everything and recreate using JS. I'm clueless about a universal solution.

When I get some more time, I will come back to this nightmare hahahaha

**Comment by MultiMadjik** [comment]
Thanks @MakD and @ToMattBan for your job, it resolve a part of the problem for me :)
Like you said, maybe find a fix solution last 

Hoever, the new CSS script are fixed also, I haven't screen problem anymore.

**Comment by brkr1** [comment]
> @ToMattBan please test this I guess solved it
> 
> ```
>     <script>
>       const createRequestTab = () => {
>         console.log("Creating request tab");
>         const title = document.createElement("div");
>         title.classList.add("emby-button-foreground");
>         title.innerText = "Requests";
>   
>         const button = document.createElement("button");
>         button.type = "button";
>         button.is = "empty-button";
>         button.classList.add("emby-tab-button", "emby-button", "lastFocused");
>         button.setAttribute("data-index", "2");
>         button.setAttribute("id", "requestTab");
>         button.appendChild(title);
>   
>         (function e() {
>           const tabb = document.querySelector(".emby-tabs-slider");
>           if (tabb && !document.querySelector("#requestTab")) {
>             tabb.appendChild(button);
>           } else if (!tabb) {
>             setTimeout(e, 500);
>           }
>         })();
>       };
>   
>       // Initial page load
>       document.addEventListener("DOMContentLoaded", () => {
>         createRequestTab();
>       });
>   
>       // When navigating back or forward
>       window.addEventListener("popstate", () => {
>         createRequestTab();
>       });
>     </script>
> ```
> 
> Edit: ARRRRGH... So
> 
> 1. When you are on a movie detail page, and press back/home, the button renders.
> 2. When you are in a library and press back it renders
> 3. When you are in a library and press home it doesn't.😂😂

Is it working on 10.10.3? It has created the tab, but shows nothing.

**Comment by Elementalzx14** [comment]
Were you able to fix the cropping issue? I am having the same issue

**Comment by Elementalzx14** [comment]
> You might want to change the iframe dimensions. That's how I did mine

So you know what needs to be changed? I tried playing with mine but couldn’t get any results, I saw no change
